lib/array.rb

```diff
@@ -0,0 +1,27 @@
+class Array
+  def my_inject(arg1 = nil, arg2 = nil)
+    return with_symbol(arg1) if arg1.is_a? Symbol
+    return with_symbol_and_init(arg1, arg2) if arg2.is_a? Symbol
+    if block_given?
+      arg1.nil? ? sum = first && shift : sum = arg1
+      each { |num| sum = yield(sum, num) }
+    end
+    sum
+  end
+  ####
+  # would like to put the if block_given? block into a seperate method
+  # Tried passing a block so that I can pass it to a different method
+  # but couldn't seem to get it to work.Moved on as short of time Advise please!
```

Comment:
Take a look at how you can use `&` for a shortcut to `to_proc`.

Your first (and second) guard clauses could therefore be
```ruby
return my_inject(&arg1) if arg1.is_a?(Symbol)
return my_inject(arg1, &arg2) if arg2
```

And you can use `yield` afterwards. See https://github.com/spike01/reinject/blob/master/lib/infect.rb for more